pkg/ruler/rulestore/local/local_test.go

```diff
@@ -69,27 +70,59 @@ func TestClient_LoadAllRuleGroups(t *testing.T) {
 	err = os.Symlink(namespace1, path.Join(dir, user1, namespace2))
 	require.NoError(t, err)
 
-	client, err := NewLocalRulesClient(Config{
+	client, err := NewLocalRulesClient(rulestore.LocalStoreConfig{
 		Directory: dir,
 	}, promRules.FileLoader{})
 	require.NoError(t, err)
 
-	ctx := context.Background()
-	userMap := map[string]rulespb.RuleGroupList{}
-
-	for _, u := range []string{user1, user2} {
-		rgl, err := client.ListRuleGroupsForUserAndNamespace(ctx, u, "") // Client loads rules in its List method.
+	t.Run("all rule groups", func(t *testing.T) {
+		ctx := context.Background()
+		userMap := map[string]rulespb.RuleGroupList{}
+
+		for _, u := range []string{user1, user2} {
```

Comment:
I don't quite understand why we're doing a loop over two users to populate a map when we could do this call in the same loop as below. Can we combine these?

Reply:
Good catch, it's indeed simpler

🆙 I updated the changes

Reply:
Thanks!